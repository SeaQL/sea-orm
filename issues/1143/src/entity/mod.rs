pub mod sea_orm_active_enums;
