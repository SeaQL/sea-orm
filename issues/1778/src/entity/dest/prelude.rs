//! `SeaORM` Entity. Generated by sea-orm-codegen 0.12.0-rc.5

pub use super::dest::Entity as Dest;
