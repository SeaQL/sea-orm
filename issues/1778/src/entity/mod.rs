pub mod dest;
pub mod src;