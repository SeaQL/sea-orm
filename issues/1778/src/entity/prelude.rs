//! SeaORM Entity. Generated by sea-orm-codegen 0.9.1

pub use super::dest::Entity as Dest;
