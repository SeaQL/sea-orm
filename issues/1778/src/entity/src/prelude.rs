//! `SeaORM` Entity. Generated by sea-orm-codegen 0.12.0-rc.4

pub use super::src::Entity as Src;
