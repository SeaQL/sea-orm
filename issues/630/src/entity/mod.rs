//! SeaORM Entity. Generated by sea-orm-codegen 0.6.0

pub mod prelude;

pub mod underscores;
pub mod underscores_workaround;
