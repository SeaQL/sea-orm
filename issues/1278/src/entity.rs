use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "pool")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
