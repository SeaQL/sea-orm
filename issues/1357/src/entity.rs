use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "pool")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: i64,
    pub name: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
