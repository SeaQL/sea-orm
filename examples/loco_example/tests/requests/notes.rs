use insta::{assert_debug_snapshot, with_settings};
use loco_rs::testing;
use migration::Migrator;
use sea_orm::entity::prelude::*;
use serial_test::serial;
use todolist::{app::App, models::_entities::notes::Entity};

// TODO: see how to dedup / extract this to app-local test utils
// not to framework, because that would require a runtime dep on insta
macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("notes_request");
        let _guard = settings.bind_to_scope();
    };
}

// Disabled integration test for GitHub CI
/*
#[tokio::test]
#[serial]
async fn can_get_notes() {
    configure_insta!();

    testing::request::<App, Migrator, _, _>(|request, ctx| async move {
        testing::seed::<App>(&ctx.db).await.unwrap();

        let notes = request.get("/api/notes").await;

        with_settings!({
            filters => {
                 let mut combined_filters = testing::CLEANUP_DATE.to_vec();
                    combined_filters.extend(vec![(r#"\"id\\":\d+"#, r#""id\":ID"#)]);
                    combined_filters
            }
        }, {
            assert_debug_snapshot!(
            (notes.status_code(), notes.text())
        );
        });
    })
    .await;
}

#[tokio::test]
#[serial]
async fn can_add_note() {
    configure_insta!();

    testing::request::<App, Migrator, _, _>(|request, _ctx| async move {
        let payload = serde_json::json!({
            "title": "loco",
            "content": "loco note test",
        });

        let add_note_request = request.post("/api/notes").json(&payload).await;

        with_settings!({
            filters => {
                 let mut combined_filters = testing::CLEANUP_DATE.to_vec();
                    combined_filters.extend(vec![(r#"\"id\\":\d+"#, r#""id\":ID"#)]);
                    combined_filters
            }
        }, {
            assert_debug_snapshot!(
            (add_note_request.status_code(), add_note_request.text())
        );
        });
    })
    .await;
}

#[tokio::test]
#[serial]
async fn can_get_note() {
    configure_insta!();

    testing::request::<App, Migrator, _, _>(|request, ctx| async move {
        testing::seed::<App>(&ctx.db).await.unwrap();

        let add_note_request = request.get("/api/notes/1").await;

        with_settings!({
            filters => {
                 let mut combined_filters = testing::CLEANUP_DATE.to_vec();
                    combined_filters.extend(vec![(r#"\"id\\":\d+"#, r#""id\":ID"#)]);
                    combined_filters
            }
        }, {
            assert_debug_snapshot!(
            (add_note_request.status_code(), add_note_request.text())
        );
        });
    })
    .await;
}

#[tokio::test]
#[serial]
async fn can_delete_note() {
    configure_insta!();

    testing::request::<App, Migrator, _, _>(|request, ctx| async move {
        testing::seed::<App>(&ctx.db).await.unwrap();

        let count_before_delete = Entity::find().all(&ctx.db).await.unwrap().len();
        let delete_note_request = request.delete("/api/notes/1").await;

        with_settings!({
            filters => {
                 let mut combined_filters = testing::CLEANUP_DATE.to_vec();
                    combined_filters.extend(vec![(r#"\"id\\":\d+"#, r#""id\":ID"#)]);
                    combined_filters
            }
        }, {
            assert_debug_snapshot!(
            (delete_note_request.status_code(), delete_note_request.text())
        );
        });

        let count_after_delete = Entity::find().all(&ctx.db).await.unwrap().len();
        assert_eq!(count_after_delete, count_before_delete - 1);
    })
    .await;
}
*/
