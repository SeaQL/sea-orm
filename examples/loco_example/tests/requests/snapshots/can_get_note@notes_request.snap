---
source: tests/requests/notes.rs
expression: "(add_note_request.status_code(), add_note_request.text())"
---
(
    200,
    "{\"created_at\":\"DATE\",\"updated_at\":\"DATE\",\"id\":ID,\"title\":\"Loco note 1\",\"content\":\"Loco note 1 content\"}",
)
